compendium/ConfigurationAdmin/src/ConfigurationImpl.hpp

```diff
@@ -118,6 +118,30 @@ class ConfigurationImpl final
    */
   void Invalidate() override;
 
+  /** Internal method used by {@code ConfigurationAdminImpl} to determine if a configuration
+   * object has been updated. 
+   *
+   * See {@code ConfigurationPrivate#HasBeenUpdatedAtLeastOnce}
+   */
+  bool HasBeenUpdatedAtLeastOnce() override { 
+    std::lock_guard<std::mutex> lk{ propertiesMutex };
+    if (changeCount > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  /** Internal method used by {@code ConfigurationAdminImpl} to get the value of the 
+   *  changeCount
+   *
+   * See {@code ConfigurationPrivate#GetChangeCount}
+   */
+  unsigned long GetChangeCount() override
+  {
+    std::lock_guard<std::mutex> lk{ propertiesMutex };
+    return changeCount;
```

Comment:
per the OSGi spec section 104.14.3.6, this function should thrown an exception if the configuration was removed. In our case you can use the standard `throw std::runtime_error(REMOVED_EXCEPTION_MESSAGE);`.